libraries/Microsoft.Bot.Builder.Core.State/ConversationStateManager.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+
+namespace Microsoft.Bot.Builder.Core.State
+{
+    public interface IConversationStateManager : IStateManager
+    {
+        string ChannelId { get; }
+        string ConversationId { get; }
+    }
+
+    public class ConversationStateManager : StateManager, IConversationStateManager
+    {
+        public static string PreferredStateStoreName = $"{nameof(ConversationStateManager)}.PreferredStateStore";
+
+        public ConversationStateManager(string channelId, string conversationId, IStateStorageProvider stateStore) : base(BuildStateNamespace(channelId, conversationId), stateStore)
+        {
+            ChannelId = channelId ?? throw new System.ArgumentNullException(nameof(channelId));
+            ConversationId = conversationId ?? throw new System.ArgumentNullException(nameof(conversationId));
+        }
+
+
+        public string ChannelId { get; }
+
+        public string ConversationId { get; }
+
+        public static string BuildStateNamespace(string channelId, string conversationId) => $"/channels/{channelId}/conversations/{conversationId}";
```

Comment:
Interesting you say this. That was something I observed a while ago, but since I've been involved I've only ever seen it called "conversation state" and figured there was some history there where the channel aspect was just implied.

Reply:
Null/Empty checks added.